components/omega/create_scripts.py

```diff
@@ -0,0 +1,375 @@
+        self.get_modules(outvar)
+        self.get_envs(outvar)
+        self.write_output(outvar)
+        self.generate_scripts(outvar)
+
+    # create a temporary cmake script to be included
+    # in the main Omega cmake build system
+    def write_output(self, outvar):
+
+        with open(self.outpath, "w") as f:
+            f.write("message(STATUS \"Reading E3SM machine info\")\n")
+
+            for key, value in outvar.items():
+                if not key.startswith("__OMEGA_"):
+                    f.write("set(ENV{%s} \"%s\")\n" % (key, value))
+
+            f.write(f"set(MACH {self.machname})\n")
+            f.write(f"set(OS {self.machos})\n")
+            f.write(f"set(COMPILER {self.compiler})\n")
+            f.write(f"set(MPI_EXEC {self.mpiexec})\n")
+            f.write(f"set(CASEROOT {self.machpath})\n")
+            f.write(f"include({self.macrospath})\n")
+
+    # create scripts
+    def generate_scripts(self, outvar):
+
+        omega_env = os.path.join(os.path.dirname(self.outpath),
+                                 "omega_env.sh")
+        omega_build = os.path.join(os.path.dirname(self.outpath),
+                                   "omega_build.sh")
+        omega_run = os.path.join(os.path.dirname(self.outpath),
+                                 "omega_run.sh")
+        omega_ctest = os.path.join(os.path.dirname(self.outpath),
+                                   "omega_ctest.sh")
+
+        with open(omega_env, "w") as f:
+            f.write("#!/usr/bin/env bash\n\n")
+
+            f.write("# module commands\n")
+            for cmd in outvar["__OMEGA_MODULE_COMMANDS__"]:
+                f.write(cmd + "\n")
+
+            f.write("\n# env. variables\n")
+            for key, value in outvar["__OMEGA_SCRIPT_EXPORTS__"].items():
+                f.write(f"export {key}=\"{value}\"\n")
+
+        with open(omega_build, "w") as f:
+            f.write("#!/usr/bin/env bash\n\n")
+
+            f.write("source ./omega_env.sh\n")
+            f.write("make\n")
+
+        with open(omega_run, "w") as f:
+            f.write("#!/usr/bin/env bash\n\n")
+
+            f.write("source ./omega_env.sh\n")
+            f.write("./src/omega.exe\n")
+
+        with open(omega_ctest, "w") as f:
+            f.write("#!/usr/bin/env bash\n\n")
+
+            f.write("source ./omega_env.sh\n")
+            f.write("ctest # --rerun-failed --output-on-failure\n")
+
+        st = os.stat(omega_env)
+        os.chmod(omega_env, st.st_mode | stat.S_IEXEC)
+
+        st = os.stat(omega_build)
+        os.chmod(omega_build, st.st_mode | stat.S_IEXEC)
+
+        st = os.stat(omega_run)
+        os.chmod(omega_run, st.st_mode | stat.S_IEXEC)
+
+        st = os.stat(omega_ctest)
+        os.chmod(omega_ctest, st.st_mode | stat.S_IEXEC)
+
+
+def main():
+
+    mach = OmegaMachines(args.cimepath, args.outpath, args.machine,
+                         args.compiler, args.debug)
+    mach.gen_machinfo()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
It seems to be complaining about too many or too few blank lines at the end of the file. There should be one plank line, depending on your editor).  However, GitHub isn't complaining about not having a newline so maybe there are too many rather than too few? No idea...

Reply:
Scratch that.  It's complaining about trailing whitespace, not new lines at the end of the file.